Node/Package managers guidelines.md

```diff
@@ -0,0 +1,282 @@
+
+**Why We Chose pnpm to avoid Doppelgangers**
+
+pnpm takes a fundamentally different approach to dependency management that directly addresses the doppelganger problem. Instead of installing dependencies in a flat or deeply nested structure, pnpm uses a content-addressable storage system where all versions of a package are stored globally on the filesystem and linked into your project’s node\_modules using hard links or symlinks.
+
+This approach offers several key advantages:
+
+* **No Redundant Installations**: With pnpm, different versions of the same package are never redundantly installed. If a version of a package is already present in the global store, it is simply linked to where it’s needed. This eliminates the doppelganger issue entirely.
+* **Consistent Dependency Resolution**: pnpm enforces a strict dependency tree that ensures only the correct and intended versions of packages are used. This minimizes the risk of conflicts that can arise from duplicate installations.
+* **Efficient Disk Usage**: By storing each version of a package only once and linking it as needed, pnpm drastically reduces the disk space used by node modules, even in large projects with many dependencies.
+* **Improved Performance**: Because pnpm avoids redundant downloads and installations, it’s faster to install dependencies, especially in projects with complex dependency trees.
+
+### Getting Started with pnpm
+
+To start using pnpm, you first need to install it globally:
+
+```bash
+npm install -g pnpm
+```
+
+You can then use pnpm just like any other package manager, but with some enhancements.
+
+#### Installing Dependencies
+
+To install all dependencies listed in your package.json, run:
+
+```bash
+pnpm install
+```
+
+#### Adding new dependencies
+
+To add a new package as a dependency, use:
+
+```bash
+pnpm add <package-name>
+```
+
+For example:
+
+```bash
+pnpm add express
+```
+
+To add development dependency:
+
+```bash
+pnpm add -D jest
+```
+
+To add dependency with exact latest version instead of one with caret:
+
+```bash
+pnpm add -E storybook
+```
+
+#### Installing dependencies
+
+To install all dependencies listed in package.json:
+
+```bash
+pnpm install
+```
+
+To install only the production dependencies (i.e., those listed under dependencies in package.json):
+
+```bash
+pnpm install --prod
+```
+
+To ensure that the pnpm-lock.yaml file is up to date and installation doesn't regenerate it, which is useful in CI environments to ensure that the lockfile is consistent with what's committed:
+
+```bash
+pnpm install --frozen-lockfile
+```
+
+To install packages from the local cache only, without trying to reach the internet:
+
+```bash
+pnpm install --offline
+```
+
+You can also combine multiple flags together, which is useful in CI/CD, for example:
+
+```bash
+pnpm install --prod --offline --frozen-lockfile
```

Comment:
`--offline` shouldn't be used with CI/CD. 
I don't think you suggest that, but as a CI/CD is mentioned in the sentence above, it could be interpreted as such.